### ISSUE TITLE
fix: regularization and early stopping — implement proper L2 + patience callback

### ISSUE THREAD

**Issue by ShaistaShabbir-prog** [ShaistaShabbir-prog/AIX#8]
## Fix: Regularization and early stopping performing poorly (#6)

Closes #6

### Current issue
The LSTM is overfitting — training accuracy >> validation accuracy.

### Fix
```python
# L2 regularization on LSTM weights
model.add(LSTM(128, kernel_regularizer=l2(0.001), recurrent_regularizer=l2(0.001)))

# Dropout between layers
model.add(Dropout(0.4))

# Early stopping with restore_best_weights
from keras.callbacks import EarlyStopping
early_stop = EarlyStopping(
    monitor='val_loss',
    patience=5,
    restore_best_weights=True,
    min_delta=0.001
)
```

Also reduce learning rate on plateau:
```python
from keras.callbacks import ReduceLROnPlateau
lr_scheduler = ReduceLROnPlateau(monitor='val_loss', factor=0.5, patience=3)
```
